### ISSUE TITLE
Standardise the anatomy of input[type=number]

### ISSUE THREAD

**Issue by lukewarlow** [openui/open-ui#769]
`<input type="number">` elements display spinners in most desktop browsers to increase/decrease the number value. These cannot be styled. There is the non-standard ::-webkit-inner-spin-button and ::-webkit-outer-spin-button, but these are limited and don't work in all browsers. Links regarding the webkit selectors:

https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-inner-spin-button
https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-outer-spin-button
https://trac.webkit.org/wiki/Styling%20Form%20Controls#inputelement

It would be good for Open UI to standardise this anatomy and then try to standardise it for better styleability.

See https://github.com/w3c/csswg-drafts/issues/8777

**Comment by lukewarlow** [comment]
This issue is mostly to bring it to this CGs attention

**Comment by gregwhitworth** [comment]
Thanks @lukewarlow - any interest in driving an explainer?

**Comment by lukewarlow** [comment]
Yeah I'm definitely interested in helping out with that I'm gonna take some time this weekend to look through the various libraries and see what they have.

**Comment by lukewarlow** [comment]
Research PR: https://github.com/openui/open-ui/pull/784

**Comment by jyasskin** [comment]
Another thought, and apologies if this is the wrong place to suggest this, or if it's achievable some other way: it would be useful to be able to set the width of the part of the input that's *not* the spinner (to make sure a certain number of digits fit), which implies it should be addressable somehow. As far as I can see, no browser currently makes this part of the control addressable.
